### PR TITLE
feat(Solid): Wave creates component

### PR DESCRIPTION
**Pull request by slhay28**
## Description

-Creates a wave component of which the animation is controlled by "toggleAnimation"
-Currently animation is controlled via chaining and setInterval to the duration of the chained animations. Once changes to renderer allow for looping of chained animations, this will be updated

## Changes

- creates whole component

## Testing

- notice how the wave works, test out different tones
- -review code

### CODE REVIEW

File: src/packages/solid/components/Wave/Wave.styles.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type { BorderStyleObject, NodeStyles } from '@lightningjs/solid';
```
Comment (anthony9187) [inline src/packages/solid/components/Wave/Wave.styles.ts:18]: remove unused import

File: src/packages/solid/components/Wave/Wave.styles.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  radius?: NodeStyles['borderRadius'];
+  itemSpacing?: NodeStyles['itemSpacing'];
+}
+
+type WaveConfig = ComponentStyleConfig<WaveStyleProperties>;
+
+/* @ts-expect-error next-line themes are supplied by client applications so this setup is necessary */
+const { Wave: { defaultTone, ...waveThemeStyles } = { styles: {} } } = theme?.componentConfig;
+/* @ts-expect-error next-line see above */
+const { Surface: { surfaceDefaultTone, ...surfaceThemeStyles } = { styles: {} } } = theme?.componentConfig;
```
Comment (anthony9187) [inline src/packages/solid/components/Wave/Wave.styles.ts:42]: the surfaceDefaultTone should be added to line 96 between defaultTone and 'neutral'

both `styles` keys should be `themeStyles

File: src/packages/solid/components/Wave/Wave.styles.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  },
+  modes: {
+    focus: {},
+    disabled: {}
+  },
+  tones: {
+    inverse: {color: theme.color.fillInverse},
+    brand: {color: theme.color.fillBrand}
+  },
+  // TODO: figure out checked state
```
Comment (anthony9187) [inline src/packages/solid/components/Wave/Wave.styles.ts:88]: checked state?

File: src/packages/solid/components/Wave/Wave.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+import type { UIComponentProps } from '../../types/interfaces.js';
+import styles from './Wave.styles.js';
+
+export interface WaveProps extends UIComponentProps {}
+
+const Wave: Component<WaveProps> = (props: WaveProps) => {
+
+  let right, left, center, rightCenter, leftCenter, Container;
+  
+  // @ts-expect-error
```
Comment (anthony9187) [inline src/packages/solid/components/Wave/Wave.tsx:29]: ts-expect-error lines require an explanatory comment

File: src/packages/solid/components/Wave/Wave.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+import { createEffect, createMemo, type Component } from 'solid-js';
+import { View } from '@lightningjs/solid';
+import type { UIComponentProps } from '../../types/interfaces.js';
+import styles from './Wave.styles.js';
+
+export interface WaveProps extends UIComponentProps {}
+
+const Wave: Component<WaveProps> = (props: WaveProps) => {
+
+  let right, left, center, rightCenter, leftCenter, Container;
```
[the remark this answers is not in the document]
Comment (anthony9187) [reply src/packages/solid/components/Wave/Wave.tsx:27]: we should add types for these

File: src/packages/solid/components/Wave/Wave.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  
+  // @ts-expect-error
+  const maxHeight = createMemo(() => props.height ?? styles.Container.tones[props.tone ?? styles.tone].height ?? styles.Container.base.height);
```
[the remark this answers is not in the document]
Comment (anthony9187) [reply src/packages/solid/components/Wave/Wave.tsx:30]: actually i think these errors might be resolved by some changes i made the Artwork PR. once that gets merged we can see, but I think we'll still need to optionally chain tones.tone.height

File: src/packages/solid/components/Wave/Wave.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    .chain({ height: (maxHeight())}, { duration: 300})
+    .chain({ height: maxHeight()/4}, { duration: 300})
+    .chain({ height: maxHeight()/4}, { duration: 50}).start();
+   }
+  }
+
+  createEffect(() => {
+    setInterval(() =>
+      startAnimation(), 1450);
+    }, isAnimating());
```
Comment (anthony9187) [inline src/packages/solid/components/Wave/Wave.tsx:88]: two things
 - is the 1450 completely just trial and error? 
 - what's isAnimating() doing here? mostly im just less familiar with the createEffect API

File: src/packages/solid/components/Wave/Wave.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+  return (
+    <View {...props} style={[props.style, styles.Container.tones?.[props.tone ?? styles.tone], styles.Container.base]} ref={Container}>
+      {/* Left */}
+      <View
```
Comment (anthony9187) [inline src/packages/solid/components/Wave/Wave.tsx:93]: these `View`s should all use the self closing syntax

```
<View props />
```

**Comment by anthony9187** [comment]
add add `@ts-expect-error` statements for each of the style and ref lines. after rebasing, you'll see my copy/pasted comment for the style tag in the other components, just reuse that. for the ref one just explain that this is expected/fine

File: src/packages/solid/components/Button/Button.styles.ts [hunk 9]
```diff
@@ -92,7 +92,7 @@ const content: ButtonConfig = {
     color: 'contentColor'
   },
   base: {
-    color: theme.color.textNeutral
+    color: theme.color.textNeutral,
```
[the remark this answers is not in the document]
Comment (slhay28) [reply src/packages/solid/components/Button/Button.styles.ts:95]: i do not even know how that got added, sorry

File: src/packages/solid/components/Wave/Wave.tsx (hunk 8 above)
Comment (slhay28) [reply src/packages/solid/components/Wave/Wave.tsx:93]: i believe they are already🤔

File: src/packages/solid/components/Wave/Wave.tsx (hunk 7 above)
Comment (slhay28) [reply src/packages/solid/components/Wave/Wave.tsx:88]: the sum of the durations is 1350, 1450 adds some buffer room so it doesn't start before the animation is completely finished.

isAnimating() is actually not needed here..oops

File: src/packages/solid/components/Wave/Wave.tsx (hunk 8 above)
Comment (anthony9187) [reply src/packages/solid/components/Wave/Wave.tsx:111]: well they weren't when i reviewed it :P

File: src/packages/solid/components/Wave/Wave.tsx [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+    leftCenter: ElementNode;
+
+  const maxHeight = createMemo(
+    () =>
+      props.height ??
+      styles.Container.tones[props.tone ?? styles.tone]?.height ??
+      styles.Container.base.height
+  );
+
+  const isAnimating = createMemo(() => props.toggleAnimation);
```
Comment (anthony9187) [inline src/packages/solid/components/Wave/Wave.tsx:38]: I think toggleAnimation needs to be added to the props interface

**Review by anthony9187 — COMMENTED** [review]
can you add comments for `WaveProps.toggleAnimation` and `WaveStyleProperties.radius`?

**Review by anthony9187 — APPROVED** [review]

**Pull request merged**